Source/UI/Test Results/IPFTestResultsManager.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+@objcMembers
+final class IPFTestResultsManager: NSObject {
+    static let shared = IPFTestResultsManager()
+
+    var results = [IPFTestResult]()
+
+    func add(_ result: IPFTestResult) {
+        results.insert(result, at: 0)
```

Comment:
Currently just saving the results in-memory. I would like to use Core Data to allow a more robust and scalable persistence, and maybe iCloud sync using NSCloudPersistentContainer (iOS 13+). Majority of the devices should be on iOS 13 and plus. To use even NSPersistenContainer, we might have to drop iOS 9 from supported devices. Even for testing the UI currently, I could only find iOS 10.3.1 simulator embedded in Xcode. Let me know what you think on this one

Reply:
Yep, I've been a bit aggressive with supporting old iOS versions, I'll need to check what device usage looks like to know if iOS 9 can go or not. For persistence, I like the idea of iCloud syncing but we'd need to introduce a device tag to know which one made the individual tests, make sure they don't step on each other's toes...

Ideally we'd have an append-only file synced over iCloud, that way we reduce the possibility of conflicts. And given tests really are read only (you only ever add them or delete them) we could completely eliminate failure cases there. I remember there was an iCloud documents API, but I forgot exactly what that is called.

All of that would involve a fair bit of complexity, so just keeping the backing store a dumb local plist for now would probably be best.

Reply:
Considering the backwards compatibility the app needs, have used a dumb backing store for now. I used json though, based on some performance tests and data sizes. Have added some database tests in the test target but below is a summary of the results.

**Large Dataset test: 10000 results**
JSON vs Plist advantages
File Size: 1.1MB vs 3.1MB - 65% lesser file size
Save: 0.009seconds vs 0.018seconds - Half the time to save
Read: 0.086 vs 0.1 seconds - 14% faster read

Also JSON is more portable and understood by non-Apple platforms.
If you still want to keep Plist encoding instead, it is a small change and we just have to switch out the encoder-decoders.